x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -217,8 +217,14 @@ private static Attribute handleSpecialFields(UnresolvedAttribute u, Attribute na
             // unsupported types
             else if (DataTypes.isUnsupported(fa.dataType())) {
                 UnsupportedEsField unsupportedField = (UnsupportedEsField) fa.field();
-                named = u.withUnresolvedMessage(
-                        "Cannot use field [" + fa.name() + "] type [" + unsupportedField.getOriginalType() + "] as is unsupported");
+                if (unsupportedField.hasInherited()) {
```

Comment:
Since the message is the same in the branch, simply do the if on the prefix
```
String inheritedMessage = "";
if (unsupportedField.hasInherited()) {
   inheritedMessage = format("in hierarchy (field [{}"], unsupportedField.getInherited()));
}
named = u.withUnresolvedMessage(... + inheritedMessage);
```
Further more replace the string concatenation in the message with `MessageFormat.format` to trim it a bit.